File: test/resources/repos/csharp/test_repo/ConcreteProcessor.cs
using System;
using System.Collections.Generic;

namespace TestRepo
{
    public class ConcreteProcessor : BaseClass, IProcessable
    {
        private List<string> data;

        public ConcreteProcessor(string name) : base(name)
        {
            data = new List<string>();
        }

        public override void Execute()
        {
            Console.WriteLine($"Executing processor {name}");
            Process();
        }

        public void Process()
        {
            Console.WriteLine($"Processing data: {string.Join(", ", data)}");
        }

        public string GetType()
        {
            return "ConcreteProcessor";
        }

        public void AddData(string item)
        {
            data.Add(item);
        }
    }
}

File: test/resources/repos/php/test_repo/ConcreteProcessor.php
<?php

require_once 'BaseClass.php';

class ConcreteProcessor extends BaseClass implements Processable 
{
    private array $data;

    public function __construct(string $name) 
    {
        parent::__construct($name);
        $this->data = [];
    }

    public function execute(): void 
    {
        echo "Executing processor {$this->name}\n";
        $this->process();
    }

    public function process(): void 
    {
        echo "Processing data: " . implode(", ", $this->data) . "\n";
    }

    public function getType(): string 
    {
        return "ConcreteProcessor";
    }

    public function addData(string $item): void 
    {
        $this->data[] = $item;
    }
}